etc/helm/pachyderm/templates/pachd/deployment.yaml

```diff
@@ -167,6 +167,18 @@ spec:
             secretKeyRef:
               name: {{ (include "pachyderm.enterpriseLicenseKeySecretName" . ) | trim | quote }}
               key: "enterprise-license-key"
+          {{- if (include "pachyderm.upstreamIDPsSecretName" .) }}
+        - name: IDP_CONNECTORS
+          valueFrom:
+            secretKeyRef:
```

Comment:
Maybe not relevant yet, but there is a limit on the total size of the environment.  I think it's 8MB, so we're probably not going to hit it, but something to be aware of if we want an unbounded amount of IDP connectors.

I think if I were implementing this right now, I would have made the environment variable point to a config file, and then mount the secret as a volume, and have pachd read the config file.  No need to change things though, just a matter of personal preference.